xstream/src/java/com/thoughtworks/xstream/core/util/Types.java

```diff
@@ -16,10 +16,16 @@
  * @since 1.4.8
  */
 public class Types {
-    private static final Pattern lambdaPattern = Pattern.compile(".*\\$\\$Lambda\\$[0-9]+/.*");
+    private static final Pattern lambdaPattern = Pattern.compile(".*\\$\\$Lambda(?:\\$[0-9]+|)/.*");
 
     public static final boolean isLambdaType(final Class<?> type) {
-        return type != null && type.isSynthetic() && lambdaPattern.matcher(type.getSimpleName()).matches();
+        if (type != null && type.isSynthetic()) {
```

Comment:
Ah the "fix" for https://bugs.openjdk.org/browse/JDK-8254979 was just change in documentation. So I think we don't need to handle Class.isSimpleName() returning empty then.